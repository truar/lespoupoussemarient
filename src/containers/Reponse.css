.Reponse {
    background: url('../images/invitation.jpg') no-repeat center fixed;
    background-size: cover; /* version standardisÃ©e */
    height: 100vh;
    align-items: center;
    color: black;
    text-shadow: none;
}

.Reponse h1 {
    color: black;
}

.Reponse>.WrapperContent {
    width: 95%;
    padding: 5px 10px;
    height: 85%;
    justify-content: space-between;
}

.Reponse>.WrapperContent>.WrapperContent>.WrapperContent, .Reponse>.WrapperContent>.WrapperContent>.WrapperContent>.WrapperContent {
    flex-direction: row;
    width: 100%;
}

.Reponse textarea.form-control {
    height: 20vh;
}

.Reponse #hebergementGroup {
    flex-direction: column;
}

.Reponse .invalid-feedback {
    text-align: center;
}

.Reponse>.WrapperContent>.WrapperContent>.WrapperContent .form-group {
    display: flex;
    align-items: center;
}

.Reponse>.WrapperContent>.WrapperContent>.WrapperContent .form-group label, .Reponse>.WrapperContent>.WrapperContent>.WrapperContent .form-group input {
    margin: 0px;
    position: relative;
}

.Reponse>.WrapperContent>.WrapperContent>.WrapperContent .form-group input {
    margin-right: .7rem;
    position: relative;
}

.Reponse .transparent-background {
    background-color: rgba(255,255,255,0.5);
    z-index: 1;
}

.Reponse .spaceDown {
    margin-bottom: 2%;
}

@media(min-width: 768px) {
    .Reponse>.WrapperContent>.WrapperContent:nth-child(2){
        width: 85%;
    }
    .Reponse .WrapperContent label {
        font-size: 1.5rem;
    }
}
@media(min-width: 992px) {
    .Reponse>.WrapperContent {
        padding: 20px 10px;
        margin-bottom: 10px;
    }
    .Reponse .WrapperContent label {
        font-size: 1.75rem;
    }
}

@media(min-width: 1440px) {
    .Reponse>.WrapperContent>.WrapperContent:nth-child(2){
        width: 70%;
    }
}