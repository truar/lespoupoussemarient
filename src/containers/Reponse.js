import React, {Component} from 'react';
import {
    Alert,
    UncontrolledAlert,
    FormFeedback,
    FormGroup,
    Label,
    Input,
  } from 'reactstrap';
import LoaderButton from '../components/LoaderButton';
import Section from '../components/Section';
import PageTitle from '../components/PageTitle';
import WrapperContent from '../components/WrapperContent';
import * as emailjs from 'emailjs-com';
import './Reponse.css';

const hebergementRadios = [
    {
        value: "chalet",
        label: "Chalet"
    },{
        value: "tente",
        label: "Tente"
    },{
        value: "gite",
        label: "Gîte"
    },{
        value: "autre",
        label: "Autre"
    }
];

class Reponse extends Component {
    constructor(props) {
        super(props);

        this.state = {
            isPresentField: {
                value: '',
                error: false
            },
            guestListField: {
                value: '',
                error: false,
            },
            hebergementField: {
                value: 'N/A',
                error: false
            },
            success: false,
            isSubmitted: false,
            loading: false
        };

        this.SERVICE_ID = process.env.REACT_APP_SERVICE_ID || "service";
        this.TEMPLATE_POSITIVE_ID = process.env.REACT_APP_TEMPLATE_POSITIVE_ID || "template_positive";
        this.TEMPLATE_NEGATIVE_ID = process.env.REACT_APP_TEMPLATE_NEGATIVE_ID || "template_negative";
        this.USER_ID = process.env.REACT_APP_USER_ID || "user";

        this.handleOptionChange = this.handleOptionChange.bind(this);
        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
        this.handleSuccess = this.handleSuccess.bind(this);
        this.handleError = this.handleError.bind(this);
        this.isFormInError = this.isFormInError.bind(this);
    }

    isFormInError = () => {
        const {isPresentField, guestListField, hebergementField} = this.state;
        const {profile} = this.props;
        
        if(isPresentField.value === '') {
            isPresentField.error = true;
        } else {
            isPresentField.error = false;
            if(guestListField.value === '') {
                guestListField.error = true;
            } else {
                guestListField.error = false;
            }
            if(profile > 1 && hebergementField.value === 'N/A' && isPresentField.value === 'true') {
                hebergementField.error = true;
            } else {
                hebergementField.error = false;
            }
        }

        this.setState({
            isPresentField: isPresentField,
            guestListField: guestListField,
            hebergementField: hebergementField
        });

        return hebergementField.error || isPresentField.error || guestListField.error;
    }

    handleOptionChange = changeEvent => {
        this.setState({
            [changeEvent.target.name + "Field"]: {
                value: changeEvent.target.value
            }
        });
    }

    handleChange = event => {
        this.setState({
          [event.target.id + "Field"]: {
                value: event.target.value
          }
        });
    }

    isEmailActivated= () => {
        return this.USER_ID !== 'user';
    }

    handleSubmit = event => {
        const isFormInvalid = this.isFormInError();
        if(isFormInvalid) {
            return ;
        }
        this.setState({
            loading: true
        });
        const {profile} = this.props;
        const invitationType = (profile === 3) ? "BRUNCH" : (profile === 2) ? "DINER" : "VIN HONNEUR";
        const {isPresentField: {value: isPresent}, guestListField: {value: guestList}, hebergementField: {value: hebergement}} = this.state;
        
        let templateParam;
        let templateId;

        if(isPresent === 'true') {
            templateParam = {
                guestList: guestList,
                hebergement: hebergement,
                invitationType: invitationType
            };
            templateId = this.TEMPLATE_POSITIVE_ID;
        } else {
            templateParam = {
                guestList: guestList,
                invitationType: invitationType
            };
            templateId = this.TEMPLATE_NEGATIVE_ID;
        }
        
        if(this.isEmailActivated()) {
            emailjs.send(this.SERVICE_ID, templateId, templateParam, this.USER_ID)
                .then((response) => {
                    this.handleSuccess();
                }, (err) => {
                    this.handleError();
                });
        } else {
            setTimeout(() => {
                this.handleSuccess();
            }, 1000);
        }
    }

    handleSuccess = () => {
        this.setState({
            success: true,
            isSubmitted: true,
            loading: false
        });
    }

    handleError = () => {
        this.setState({
            success: false,
            isSubmitted: true,
            loading: false
        });
    }

    getEndForm = () => {
        const {success, isSubmitted, loading} = this.state;
        const button = (<LoaderButton
          type="submit"
          text="Envoyer"
          isLoading={loading}
          loadingText="Envoi en cours..."
          onClick={this.handleSubmit}
        />);
        let endForm = '';
        if(success === true && isSubmitted === true) {
            endForm = (<Alert color="success">
                Votre réponse a bien été envoyée. Merci !
            </Alert>);
        } else if(success === false && isSubmitted === true) {
            endForm = (<><UncontrolledAlert color="danger">
                Une erreur s'est produite.
            </UncontrolledAlert>
            {button}
            </>);
        } else {
            endForm = button;
        }
        return endForm;
    }

    getSecondPart = () => {
        const {isPresentField: {value: isPresent}, 
            guestListField: {value: guestList, error: guestListError}, 
            hebergementField: {value: hebergement, error: hebergementError}, 
            success} = this.state;
        let secondPart = '';
        const hebergementDisplay = hebergementRadios.map((radio) => {
            return (
                <FormGroup key={radio.value} >
                    <Input id={radio.value + "Id"} disabled={success} 
                        type="radio" name="hebergement" value={radio.value}
                            onChange={this.handleOptionChange}
                            checked={hebergement === radio.value} />   
                    <Label htmlFor={radio.value + "Id"}>{radio.label}</Label>
                </FormGroup>
            );
        });
        if(isPresent === 'true') {
            secondPart = (<>
                <WrapperContent>
                    <p className="spaceDown">Veuillez renseigner la liste des personnes qui seront présentes ainsi que toutes informations dont vous souhaiteriez nous faire part.</p>
                    <Input invalid={guestListError} disabled={success} type="textarea" name="guestList" id="guestList" placeholder="Prénom nom" 
                        value={guestList} onChange={this.handleChange}/>
                </WrapperContent>
                {this.props.profile > 1 &&
                    <WrapperContent>
                        <p className="spaceDown">Merci de nous indiquer le type de logement que vous souhaiteriez avoir pour la nuit.</p>
                        <WrapperContent id="hebergementGroup">
                            <WrapperContent>
                                {hebergementDisplay}
                            </WrapperContent>
                            {hebergementError && <FormFeedback style={{display: 'block'}}>Veuillez sélectionner un hébergement</FormFeedback>}
                        </WrapperContent>
                    </WrapperContent>
                }
            </>);
        } else if(isPresent === 'false') {
            secondPart = (
                <WrapperContent>
                    <p className="spaceDown">Merci de renseigner votre nom et prénom.</p>
                    <Input invalid={guestListError} disabled={success} type="text" name="guestList" id="guestList" placeholder="Nom prénom" 
                        value={guestList} onChange={this.handleChange}/>
                </WrapperContent>
            );
        }
        return secondPart;
    }

    render() {
        const {isPresentField: {value: isPresent, error: isPresentError}, success} = this.state;
        let endForm = this.getEndForm();
        let secondPart = this.getSecondPart();

        return (
            <Section id="Reponse" className="Reponse" {...this.props}>
                <PageTitle title="Répondre à l'invitation" />
                <WrapperContent>
                    <WrapperContent>
                        <p>Serez-vous diponible le Jour-J ?</p>
                        <WrapperContent>
                            <FormGroup>
                                <Input id="yesInput" disabled={success} type="radio" name="isPresent" value={true}
                                onChange={this.handleOptionChange}
                                checked={isPresent === 'true'} />
                                <Label htmlFor="yesInput">Oui</Label>
                            </FormGroup>
                            <FormGroup>
                                <Input id="noInput" disabled={success} type="radio" name="isPresent" value={false}
                                    onChange={this.handleOptionChange}
                                    checked={isPresent === 'false'} />
                                <Label htmlFor="noInput">Non</Label>
                            </FormGroup>
                        </WrapperContent>
                        {isPresentError && <FormFeedback style={{display: 'block'}}>Veuillez sélectionner une réponse</FormFeedback>}
                    </WrapperContent>
                    {secondPart}
                <WrapperContent>
                    {endForm}
                </WrapperContent>
                </WrapperContent>
            </Section>
        );
    }
}

export default Reponse;