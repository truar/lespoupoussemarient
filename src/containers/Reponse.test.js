import React from 'react';
import Reponse from './Reponse';
import { shallow } from 'enzyme';

it('renders without crashing', () => {
  const component = shallow(<Reponse />);
  expect(component).toMatchSnapshot();
});
