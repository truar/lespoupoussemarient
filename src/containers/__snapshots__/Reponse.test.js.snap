// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Reponse />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <PageTitle
          title="Répondre à l'invitation"
        />,
        <WrapperContent>
          <WrapperContent>
            <p>
              Serez-vous diponible le Jour-J ?
            </p>
            <WrapperContent>
              <FormGroup
                tag="div"
              >
                <Input
                  checked={false}
                  disabled={false}
                  id="yesInput"
                  name="isPresent"
                  onChange={[Function]}
                  type="radio"
                  value={true}
                />
                <Label
                  htmlFor="yesInput"
                  tag="label"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ]
                  }
                >
                  Oui
                </Label>
              </FormGroup>
              <FormGroup
                tag="div"
              >
                <Input
                  checked={false}
                  disabled={false}
                  id="noInput"
                  name="isPresent"
                  onChange={[Function]}
                  type="radio"
                  value={false}
                />
                <Label
                  htmlFor="noInput"
                  tag="label"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ]
                  }
                >
                  Non
                </Label>
              </FormGroup>
            </WrapperContent>
          </WrapperContent>
          
          <WrapperContent>
            <_default
              isLoading={false}
              loadingText="Envoi en cours..."
              onClick={[Function]}
              text="Envoyer"
              type="submit"
            />
          </WrapperContent>
        </WrapperContent>,
      ],
      "className": "Reponse",
      "id": "Reponse",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "title": "Répondre à l'invitation",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <WrapperContent>
              <p>
                Serez-vous diponible le Jour-J ?
              </p>
              <WrapperContent>
                <FormGroup
                  tag="div"
                >
                  <Input
                    checked={false}
                    disabled={false}
                    id="yesInput"
                    name="isPresent"
                    onChange={[Function]}
                    type="radio"
                    value={true}
                  />
                  <Label
                    htmlFor="yesInput"
                    tag="label"
                    widths={
                      Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                    }
                  >
                    Oui
                  </Label>
                </FormGroup>
                <FormGroup
                  tag="div"
                >
                  <Input
                    checked={false}
                    disabled={false}
                    id="noInput"
                    name="isPresent"
                    onChange={[Function]}
                    type="radio"
                    value={false}
                  />
                  <Label
                    htmlFor="noInput"
                    tag="label"
                    widths={
                      Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                    }
                  >
                    Non
                  </Label>
                </FormGroup>
              </WrapperContent>
            </WrapperContent>,
            "",
            <WrapperContent>
              <_default
                isLoading={false}
                loadingText="Envoi en cours..."
                onClick={[Function]}
                text="Envoyer"
                type="submit"
              />
            </WrapperContent>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <p>
                  Serez-vous diponible le Jour-J ?
                </p>,
                <WrapperContent>
                  <FormGroup
                    tag="div"
                  >
                    <Input
                      checked={false}
                      disabled={false}
                      id="yesInput"
                      name="isPresent"
                      onChange={[Function]}
                      type="radio"
                      value={true}
                    />
                    <Label
                      htmlFor="yesInput"
                      tag="label"
                      widths={
                        Array [
                          "xs",
                          "sm",
                          "md",
                          "lg",
                          "xl",
                        ]
                      }
                    >
                      Oui
                    </Label>
                  </FormGroup>
                  <FormGroup
                    tag="div"
                  >
                    <Input
                      checked={false}
                      disabled={false}
                      id="noInput"
                      name="isPresent"
                      onChange={[Function]}
                      type="radio"
                      value={false}
                    />
                    <Label
                      htmlFor="noInput"
                      tag="label"
                      widths={
                        Array [
                          "xs",
                          "sm",
                          "md",
                          "lg",
                          "xl",
                        ]
                      }
                    >
                      Non
                    </Label>
                  </FormGroup>
                </WrapperContent>,
                false,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Serez-vous diponible le Jour-J ?",
                },
                "ref": null,
                "rendered": "Serez-vous diponible le Jour-J ?",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <FormGroup
                      tag="div"
                    >
                      <Input
                        checked={false}
                        disabled={false}
                        id="yesInput"
                        name="isPresent"
                        onChange={[Function]}
                        type="radio"
                        value={true}
                      />
                      <Label
                        htmlFor="yesInput"
                        tag="label"
                        widths={
                          Array [
                            "xs",
                            "sm",
                            "md",
                            "lg",
                            "xl",
                          ]
                        }
                      >
                        Oui
                      </Label>
                    </FormGroup>,
                    <FormGroup
                      tag="div"
                    >
                      <Input
                        checked={false}
                        disabled={false}
                        id="noInput"
                        name="isPresent"
                        onChange={[Function]}
                        type="radio"
                        value={false}
                      />
                      <Label
                        htmlFor="noInput"
                        tag="label"
                        widths={
                          Array [
                            "xs",
                            "sm",
                            "md",
                            "lg",
                            "xl",
                          ]
                        }
                      >
                        Non
                      </Label>
                    </FormGroup>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": Array [
                        <Input
                          checked={false}
                          disabled={false}
                          id="yesInput"
                          name="isPresent"
                          onChange={[Function]}
                          type="radio"
                          value={true}
                        />,
                        <Label
                          htmlFor="yesInput"
                          tag="label"
                          widths={
                            Array [
                              "xs",
                              "sm",
                              "md",
                              "lg",
                              "xl",
                            ]
                          }
                        >
                          Oui
                        </Label>,
                      ],
                      "tag": "div",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "checked": false,
                          "disabled": false,
                          "id": "yesInput",
                          "name": "isPresent",
                          "onChange": [Function],
                          "type": "radio",
                          "value": true,
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "function",
                        "props": Object {
                          "children": "Oui",
                          "htmlFor": "yesInput",
                          "tag": "label",
                          "widths": Array [
                            "xs",
                            "sm",
                            "md",
                            "lg",
                            "xl",
                          ],
                        },
                        "ref": null,
                        "rendered": "Oui",
                        "type": [Function],
                      },
                    ],
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": Array [
                        <Input
                          checked={false}
                          disabled={false}
                          id="noInput"
                          name="isPresent"
                          onChange={[Function]}
                          type="radio"
                          value={false}
                        />,
                        <Label
                          htmlFor="noInput"
                          tag="label"
                          widths={
                            Array [
                              "xs",
                              "sm",
                              "md",
                              "lg",
                              "xl",
                            ]
                          }
                        >
                          Non
                        </Label>,
                      ],
                      "tag": "div",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "checked": false,
                          "disabled": false,
                          "id": "noInput",
                          "name": "isPresent",
                          "onChange": [Function],
                          "type": "radio",
                          "value": false,
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "function",
                        "props": Object {
                          "children": "Non",
                          "htmlFor": "noInput",
                          "tag": "label",
                          "widths": Array [
                            "xs",
                            "sm",
                            "md",
                            "lg",
                            "xl",
                          ],
                        },
                        "ref": null,
                        "rendered": "Non",
                        "type": [Function],
                      },
                    ],
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              false,
            ],
            "type": [Function],
          },
          "",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <_default
                isLoading={false}
                loadingText="Envoi en cours..."
                onClick={[Function]}
                text="Envoyer"
                type="submit"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "isLoading": false,
                "loadingText": "Envoi en cours...",
                "onClick": [Function],
                "text": "Envoyer",
                "type": "submit",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <PageTitle
            title="Répondre à l'invitation"
          />,
          <WrapperContent>
            <WrapperContent>
              <p>
                Serez-vous diponible le Jour-J ?
              </p>
              <WrapperContent>
                <FormGroup
                  tag="div"
                >
                  <Input
                    checked={false}
                    disabled={false}
                    id="yesInput"
                    name="isPresent"
                    onChange={[Function]}
                    type="radio"
                    value={true}
                  />
                  <Label
                    htmlFor="yesInput"
                    tag="label"
                    widths={
                      Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                    }
                  >
                    Oui
                  </Label>
                </FormGroup>
                <FormGroup
                  tag="div"
                >
                  <Input
                    checked={false}
                    disabled={false}
                    id="noInput"
                    name="isPresent"
                    onChange={[Function]}
                    type="radio"
                    value={false}
                  />
                  <Label
                    htmlFor="noInput"
                    tag="label"
                    widths={
                      Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                    }
                  >
                    Non
                  </Label>
                </FormGroup>
              </WrapperContent>
            </WrapperContent>
            
            <WrapperContent>
              <_default
                isLoading={false}
                loadingText="Envoi en cours..."
                onClick={[Function]}
                text="Envoyer"
                type="submit"
              />
            </WrapperContent>
          </WrapperContent>,
        ],
        "className": "Reponse",
        "id": "Reponse",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "title": "Répondre à l'invitation",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <WrapperContent>
                <p>
                  Serez-vous diponible le Jour-J ?
                </p>
                <WrapperContent>
                  <FormGroup
                    tag="div"
                  >
                    <Input
                      checked={false}
                      disabled={false}
                      id="yesInput"
                      name="isPresent"
                      onChange={[Function]}
                      type="radio"
                      value={true}
                    />
                    <Label
                      htmlFor="yesInput"
                      tag="label"
                      widths={
                        Array [
                          "xs",
                          "sm",
                          "md",
                          "lg",
                          "xl",
                        ]
                      }
                    >
                      Oui
                    </Label>
                  </FormGroup>
                  <FormGroup
                    tag="div"
                  >
                    <Input
                      checked={false}
                      disabled={false}
                      id="noInput"
                      name="isPresent"
                      onChange={[Function]}
                      type="radio"
                      value={false}
                    />
                    <Label
                      htmlFor="noInput"
                      tag="label"
                      widths={
                        Array [
                          "xs",
                          "sm",
                          "md",
                          "lg",
                          "xl",
                        ]
                      }
                    >
                      Non
                    </Label>
                  </FormGroup>
                </WrapperContent>
              </WrapperContent>,
              "",
              <WrapperContent>
                <_default
                  isLoading={false}
                  loadingText="Envoi en cours..."
                  onClick={[Function]}
                  text="Envoyer"
                  type="submit"
                />
              </WrapperContent>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <p>
                    Serez-vous diponible le Jour-J ?
                  </p>,
                  <WrapperContent>
                    <FormGroup
                      tag="div"
                    >
                      <Input
                        checked={false}
                        disabled={false}
                        id="yesInput"
                        name="isPresent"
                        onChange={[Function]}
                        type="radio"
                        value={true}
                      />
                      <Label
                        htmlFor="yesInput"
                        tag="label"
                        widths={
                          Array [
                            "xs",
                            "sm",
                            "md",
                            "lg",
                            "xl",
                          ]
                        }
                      >
                        Oui
                      </Label>
                    </FormGroup>
                    <FormGroup
                      tag="div"
                    >
                      <Input
                        checked={false}
                        disabled={false}
                        id="noInput"
                        name="isPresent"
                        onChange={[Function]}
                        type="radio"
                        value={false}
                      />
                      <Label
                        htmlFor="noInput"
                        tag="label"
                        widths={
                          Array [
                            "xs",
                            "sm",
                            "md",
                            "lg",
                            "xl",
                          ]
                        }
                      >
                        Non
                      </Label>
                    </FormGroup>
                  </WrapperContent>,
                  false,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Serez-vous diponible le Jour-J ?",
                  },
                  "ref": null,
                  "rendered": "Serez-vous diponible le Jour-J ?",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": Array [
                      <FormGroup
                        tag="div"
                      >
                        <Input
                          checked={false}
                          disabled={false}
                          id="yesInput"
                          name="isPresent"
                          onChange={[Function]}
                          type="radio"
                          value={true}
                        />
                        <Label
                          htmlFor="yesInput"
                          tag="label"
                          widths={
                            Array [
                              "xs",
                              "sm",
                              "md",
                              "lg",
                              "xl",
                            ]
                          }
                        >
                          Oui
                        </Label>
                      </FormGroup>,
                      <FormGroup
                        tag="div"
                      >
                        <Input
                          checked={false}
                          disabled={false}
                          id="noInput"
                          name="isPresent"
                          onChange={[Function]}
                          type="radio"
                          value={false}
                        />
                        <Label
                          htmlFor="noInput"
                          tag="label"
                          widths={
                            Array [
                              "xs",
                              "sm",
                              "md",
                              "lg",
                              "xl",
                            ]
                          }
                        >
                          Non
                        </Label>
                      </FormGroup>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "children": Array [
                          <Input
                            checked={false}
                            disabled={false}
                            id="yesInput"
                            name="isPresent"
                            onChange={[Function]}
                            type="radio"
                            value={true}
                          />,
                          <Label
                            htmlFor="yesInput"
                            tag="label"
                            widths={
                              Array [
                                "xs",
                                "sm",
                                "md",
                                "lg",
                                "xl",
                              ]
                            }
                          >
                            Oui
                          </Label>,
                        ],
                        "tag": "div",
                      },
                      "ref": null,
                      "rendered": Array [
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "class",
                          "props": Object {
                            "checked": false,
                            "disabled": false,
                            "id": "yesInput",
                            "name": "isPresent",
                            "onChange": [Function],
                            "type": "radio",
                            "value": true,
                          },
                          "ref": null,
                          "rendered": null,
                          "type": [Function],
                        },
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "function",
                          "props": Object {
                            "children": "Oui",
                            "htmlFor": "yesInput",
                            "tag": "label",
                            "widths": Array [
                              "xs",
                              "sm",
                              "md",
                              "lg",
                              "xl",
                            ],
                          },
                          "ref": null,
                          "rendered": "Oui",
                          "type": [Function],
                        },
                      ],
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "children": Array [
                          <Input
                            checked={false}
                            disabled={false}
                            id="noInput"
                            name="isPresent"
                            onChange={[Function]}
                            type="radio"
                            value={false}
                          />,
                          <Label
                            htmlFor="noInput"
                            tag="label"
                            widths={
                              Array [
                                "xs",
                                "sm",
                                "md",
                                "lg",
                                "xl",
                              ]
                            }
                          >
                            Non
                          </Label>,
                        ],
                        "tag": "div",
                      },
                      "ref": null,
                      "rendered": Array [
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "class",
                          "props": Object {
                            "checked": false,
                            "disabled": false,
                            "id": "noInput",
                            "name": "isPresent",
                            "onChange": [Function],
                            "type": "radio",
                            "value": false,
                          },
                          "ref": null,
                          "rendered": null,
                          "type": [Function],
                        },
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "function",
                          "props": Object {
                            "children": "Non",
                            "htmlFor": "noInput",
                            "tag": "label",
                            "widths": Array [
                              "xs",
                              "sm",
                              "md",
                              "lg",
                              "xl",
                            ],
                          },
                          "ref": null,
                          "rendered": "Non",
                          "type": [Function],
                        },
                      ],
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
                false,
              ],
              "type": [Function],
            },
            "",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <_default
                  isLoading={false}
                  loadingText="Envoi en cours..."
                  onClick={[Function]}
                  text="Envoyer"
                  type="submit"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "isLoading": false,
                  "loadingText": "Envoi en cours...",
                  "onClick": [Function],
                  "text": "Envoyer",
                  "type": "submit",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
